src/python_routing_v02/troute/routing/diffusive_utils.py

```diff
@@ -387,6 +388,133 @@ def fp_dbcd_map(usgsID2tw=None, usgssDT=None, usgseDT=None, usgspCd=None):
+                for seg in range(0,ncomp):
+                    if seg==ncomp-1 and x > 0:
+                        #In node-configuration, bottom node takes bathy of the first segment of the downtream reach
+                        # except TW reach where the bottom node bathy is interpolated by bathy of the last segment 
+                        # with so*0.5*dx of 
+                        segID= reach["downstream_head_segment"]
+                    else:
+                        # In node configuration, for example, for 2 segments of a reach, 
+                        # 1st node(=top node) takes bathy of the first segment and
+                        # 2nd node takes bathy of the second segment
+                        # 3rd node takes bathy of, as explained right above, the firt segment 
+                        segID= seg_list[seg]
+                    
+                     ## **** all hypothetical coding
+                    if x==2 and seg==0:
+                        idx_comid=0
+                    if x==2 and seg==1:
+                        idx_comid=1                        
+                    if x==1 and seg==0:
+                        idx_comid=1
+                    if x==1 and seg==1:
+                        idx_comid=2
+                    if x==0 and seg==0:
+                        idx_comid=2
+                    if x==0 and seg==1:
+                        #when tail water node, z_bathy will be adjusted by so*dx of related segment
+                        idx_comid=2                                      
+                    
+                    inland_bathyDF_subset = inland_bathyDF[inland_bathyDF['comid'] == comid_bathy[idx_comid]]
+                    size_bathy_g[seg, frj]= inland_bathyDF_subset['x_bathy'].size
+                    
+                    for idp in range(0, size_bathy_g[seg, frj]):
+                        x_bathy_g[idp, seg, frj]  = inland_bathyDF_subset [['x_bathy']].iloc[idp]
+                        z_bathy_g[idp, seg, frj] =  inland_bathyDF_subset [['z_bathy']].iloc[idp]
+                        mann_bathy_g[idp, seg, frj] = inland_bathyDF_subset [['n']].iloc[idp]
+                        
+                if seg == ncomp - 1 and seg_list.count(dbfksegID) > 0:
+                    # At tailwater bottom node (as # of segment = # node +1), z_bathy need to be adjusted by so*dx of the last segment
+                    segID2 = seg_list[seg - 1]
+                    #idx_segID2 = np.where(geo_index == segID2)
+                    #idx_so = np.where(geo_cols == "s0")
+                    #idx_dx = np.where(geo_cols == "dx")
+                    idx_segID2 = np.where(param_df.index.values == segID2)
+                    idx_so = np.where(param_df.columns.values == "s0")
+                    idx_dx = np.where(param_df.columns.values == "dx")
+                    
+                    #So = geo_data[idx_segID2, idx_so]
+                    #dx = geo_data[idx_segID2, idx_dx]
```

Comment:
commented lined can be cleaned out. We are no longer using geo_data numpy arrays and sticking with param_df pandas dataframe